bots/pr/src/main/java/org/openjdk/skara/bots/pr/BackportCommand.java

```diff
@@ -52,20 +57,100 @@ public boolean allowedInCommit() {
+        }
+
+        var args = command.args();
+        if (args.isBlank()) {
+            showHelpInPR(reply);
+            return;
+        }
+
+        var parts = args.split(" ");
+        boolean argIsValid = parts[0].equals("disable") ? parts.length == 2 || parts.length == 3 : parts.length <= 2;
+        if (!argIsValid) {
+            showHelpInPR(reply);
+            return;
+        }
+
+        if (parts[0].equals("disable")) {
+            // Remove label
+            var targetRepoName = parts[1];
+            var targetBranchName = parts.length == 3 ? parts[2] : "master";
+            var backportLabel = generateBackportLabel(targetRepoName, targetBranchName);
+            if (pr.labelNames().contains(backportLabel)) {
+                labelsToRemove.add(backportLabel);
+                reply.println("Backport for repo `" + targetRepoName + "` on branch `" + targetBranchName + "` was successfully disabled.");
+            } else {
+                reply.println("Backport for repo `" + targetRepoName + "` on branch `" + targetBranchName + "` was not enabled, so you can not disable it.");
+            }
+        } else {
+            // Add label
+            var forge = bot.repo().forge();
+            var repoNameArg = parts[0].replace("http://", "")
+                    .replace("https://", "")
+                    .replace(forge.hostname() + "/", "");
+            // If the arg is given with a namespace prefix, look for an exact match,
+            // otherwise cut off the namespace prefix before comparing with the forks
+            // config.
+            var includesNamespace = repoNameArg.contains("/");
+            var repoNameOptional = bot.forks().keySet().stream()
+                    .filter(s -> includesNamespace
+                            ? s.equals(repoNameArg)
+                            : s.substring(s.indexOf("/") + 1).equals(repoNameArg))
+                    .findAny();
+            String targetRepoName = repoNameOptional.orElse("<not found>");
+
+            var potentialTargetRepo = repoNameOptional.flatMap(forge::repository);
+            if (potentialTargetRepo.isEmpty()) {
+                reply.println("The target repository `" + repoNameArg + "` is not a valid target for backports. ");
+                reply.print("List of valid target repositories: ");
+                reply.println(String.join(", ", bot.forks().keySet().stream().sorted().toList()) + ".");
+                reply.println("Supplying the organization/group prefix is optional.");
+                return;
+            }
+            var targetRepo = potentialTargetRepo.get();
+
+            var targetBranchName = parts.length == 2 ? parts[1] : "master";
+            var targetBranches = targetRepo.branches();
+            if (targetBranches.stream().noneMatch(b -> b.name().equals(targetBranchName))) {
+                reply.println("The target branch `" + targetBranchName + "` does not exist");
+                return;
+            }
```

Comment:
This looks like it has been copied from the other method. Would be better to break it out into methods and reused.

Reply:
Yes, It's copied from the other method. I tried to extract the code into a method but found it's complex. I will have a try later.